modules/react/button/stories/button/examples/HyperlinkExternal.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+import {Hyperlink} from '@workday/canvas-kit-react/button';
+import {HStack} from '@workday/canvas-kit-labs-react/layout';
+import {SystemIcon} from '@workday/canvas-kit-react/icon';
+import {extLinkIcon} from '@workday/canvas-system-icons-web';
+
+export const LinkExternal = () => (
+  <HStack
+    as={Hyperlink}
+    target="_blank"
+    href="http://workday.com"
+    rel="noreferrer"
+    display="inline-flex"
+    alignItems="center"
+    spacing="xxxs"
+  >
+    <span>External HyperLink</span>
+    <SystemIcon icon={extLinkIcon} aria-label="Opens link in new window" size={16} />
```

Comment:
Design question, should we should how to make the icon color match the text color
```
const [hover, setHover] = React.useState(false)
...

<HStack
    as={Hyperlink}
    ...
    onMouseOut={() => setHover(false)}
    onMouseOver={() => setHover(true)}
    onBlur={() => setHover(false)}
    onFocus={() => setHover(true)}
>
     ...
    <SystemIcon
        icon={extLinkIcon}
        ...
        color={hover ? colors.blueberry500 : colors.blueberry400}
        colorHover={colors.blueberry500}
    />
</HStack>
```

Reply:
That's a good question. And assuming yes, what's the best way to do that? It could be done via CSS selectors or by state like in your example. I tend to not like changing state when DOM state changes when CSS is capable of that. Emotion handles this fairly well, but the CSS selectors can get a bit tricky. Your example causes inefficiencies because the React reconciler and Emotion both get involved: https://codesandbox.io/s/external-link-css-differences-ly3mk

A profile of a mouse over on the React state (your suggested version):
<img width="487" alt="image" src="https://user-images.githubusercontent.com/338257/150422244-036845e3-6158-400f-92d4-dce3b39cc0e5.png">

A profile of a mouse over on the pre-calculated CSS version:

<img width="478" alt="image" src="https://user-images.githubusercontent.com/338257/150425116-5d5f4f97-dcca-4a44-845b-a4d2e70fe158.png">

You can see both screenshots have a `mouseout` and a `mouseover`. The first one has a larger `mouseover` with a `Recalculate Style` block within the `mouseover` event because Emotion has to inject a new stylesheet which can cause layout cache invalidation. It is hard to see from the screenshot, but the first one contains 2 style recalculations. The first being from the browser and the second being triggered from a change in Emotion. The second example does not require a style recalculation for the hover because no styles have change. Both contain the usual paint and composition.

Reply:
Perhaps these nuances warrant an `ExternalHyperlink` component?

Reply:
ExternalHyperlink makes a lot of sense to me